server/src/test/java/org/apache/druid/client/indexing/NoopOverlordClient.java

```diff
@@ -84,7 +85,7 @@ public ListenableFuture<TaskPayloadResponse> taskPayload(String taskId)
   }
 
   @Override
-  public ListenableFuture<Map<String, Object>> taskReportAsMap(String taskId)
+  public ListenableFuture<TaskReport.ReportMap> taskReportAsMap(String taskId)
```

Comment:
I'd suggest adding a simple serde test that verifies that the old and new objects being returned roundtrip fine for compatibility reasons (similar to this comment https://github.com/apache/druid/pull/16217#discussion_r1545912070).

Reply:
Added.